maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/AbstractClasspathForkConfiguration.java

```diff
@@ -37,18 +37,18 @@
 {
 
     @SuppressWarnings( "checkstyle:parameternumber" )
-    public AbstractClasspathForkConfiguration( @Nonnull Classpath bootClasspath,
-                                               @Nonnull File tempDirectory,
-                                               @Nullable String debugLine,
-                                               @Nonnull File workingDirectory,
-                                               @Nonnull Properties modelProperties,
-                                               @Nullable String argLine,
-                                               @Nonnull Map<String, String> environmentVariables,
-                                               boolean debug,
-                                               int forkCount,
-                                               boolean reuseForks,
-                                               @Nonnull Platform pluginPlatform,
-                                               @Nonnull ConsoleLogger log )
+    AbstractClasspathForkConfiguration( @Nonnull Classpath bootClasspath,
```

Comment:
why removing public constructor this is not backward compat???? if checkstyle complains about this so remove checkstyle usage as it's useless....

Reply:
This constructor wasn't used outside the package. Intellij also confirms it, so checkstyle reports it properly. Also keep in mind that tests passes. If it not being used, access can be safly lowered to package level.

Reply:
I don't buy that. Those are some internal classes, not a part of any API. I fact I don't think that any part of surefire-plugin is a API. Is there rely some projects that rely on surefire internal classes? 

I think that eventual API classes and interfaces should be marked strictly as API and kept backward compatible. But not internal classes.

Reply:
I guess you're right sorry I didn't check class declaration. so yes that will work. 
So let's go for checkstyle driven development..... :-)